git.adoc

```diff
@@ -0,0 +1,211 @@
+= Git guidelines
+Jakub Jirutka
+// URIs
+:uri-git-scm: https://git-scm.com
+
+This guide aims to provide the ground rules and recommendations for using {uri-git-scm}[git] in software projects with releases.
+
+The described rules are based on widely recognized best-practices for using git (see <<References, references>>).
+The workflow is based on https://docs.gitlab.com/ee/workflow/gitlab_flow.html[GitLab Flow].
+
+To keep it short and easy to follow, it does not explain reasons for particular rules and alternative approaches.
+If you’re in doubt, consult articles in the <<References>> section — they explain most of the concepts into a great detail.
+
+
+== General
+
+=== Commit content
+
+. Every commit should be atomic.
++
+____
+atomic: of or forming a single irreducible unit or component in a larger system
+____
++
+TODO
+
+. Do not commit generated files.
++
+TODO
+
+. Do not copy third-party files, if they can be reliably obtained from some repository.
++
+Prefer dependency management tools (e.g. Bundler, Cargo, npm) to fetch project dependencies instead of copying them into your repository.
+If this option is not available, but the third-party code is hosted in a git repository, use {uri-git-scm}/docs/git-submodule[git-submodule] or {uri-git-scm}/book/en/v1/Git-Tools-Subtree-Merging[git-subtree].
```

Comment:
What if third-party code needs just small change and/or is not maintained anymore ? Shall we introduce patch sets or is it reason to copy&modify or what is right approach ?